esti/catalog_export_test.go

```diff
@@ -0,0 +1,418 @@
+	TableDescriptorPath string
+	GlueScriptPath      string
+	Branch              string
+	GlueDB              string
+	AccessKeyId         string
+	SecretAccessKey     string
+	OverrideCommitID    string
+	TableSpec           *hiveTableSpec
+}
+
+const glueExportScript = `
+local aws = require("aws")
+local exporter = require("lakefs/catalogexport/glue_exporter")
+action.commit_id = "{{ .OverrideCommitID }}" -- override commit id to use specific symlink file previously created
+local glue = aws.glue_client(args.aws.aws_access_key_id, args.aws.aws_secret_access_key, args.aws.aws_region)
+exporter.export_glue(glue, args.catalog.db_name, args.table_source, args.catalog.table_input, action, {debug=true})
+`
+
+const glueExporterAction = `
+name: Glue Exporter
+on:
+  post-commit:
+    branches: ["{{ .Branch }}*"]
+hooks:
+  - id: glue_exporter
+    type: lua
+    properties:
+      script_path: "{{ .GlueScriptPath }}"
+      args:
+        aws:
+          aws_access_key_id: "{{ .AccessKeyId }}"
+          aws_secret_access_key: "{{ .SecretAccessKey }}"
+          aws_region: us-east-1
+        table_source: '{{ .TableDescriptorPath }}'
+        catalog:
+          db_name: "{{ .GlueDB }}"
+          table_input:
+            StorageDescriptor: 
+              InputFormat: "org.apache.hadoop.hive.ql.io.SymlinkTextInputFormat"
+              OutputFormat: "org.apache.hadoop.hive.ql.io.HiveIgnoreKeyTextOutputFormat"
+              SerdeInfo:
+                SerializationLibrary: "org.apache.hadoop.hive.serde2.OpenCSVSerde"
+                Parameters:
+                  separatorChar: ","
+            Parameters: 
+              classification: "csv"
+              "skip.header.line.count": "1"`
+
+const symlinkExporterScript = `
+local exporter = require("lakefs/catalogexport/symlink_exporter")
+local aws = require("aws")
+local table_path = args.table_source
+local s3 = aws.s3_client(args.aws.aws_access_key_id, args.aws.aws_secret_access_key, args.aws.aws_region)
+exporter.export_s3(s3, table_path, action, {debug=true})
+`
+
+const symlinkExporterAction = `
+name: Symlink S3 Exporter
+on:
+  post-commit:
+    branches: ["{{ .Branch }}*"]
+hooks:
+  - id: symlink_exporter
+    type: lua
+    properties:
+      script_path: "{{ .SymlinkScriptPath }}"
+      args:
+        aws:
+          aws_access_key_id: "{{ .AccessKeyId }}"
+          aws_secret_access_key: "{{ .SecretAccessKey }}"
+          aws_region: us-east-1
+        table_source: '{{ .TableDescriptorPath }}'`
+
+func renderTplAsStr(t *testing.T, tplData any, name, content string) string {
+	t.Helper()
+	tpl, err := template.New(name).Parse(content)
+	require.NoError(t, err, "rendering template")
+	var doc bytes.Buffer
+	err = tpl.Execute(&doc, tplData)
+	require.NoError(t, err)
+	return doc.String()
+}
+
+func setupGlueClient(ctx context.Context, accessKeyID, secretAccessKey string) (*glue.Client, error) {
+	cfg, err := config.LoadDefaultConfig(ctx,
+		config.WithRegion("us-east-1"),
```

Comment:
I think we have an envariable with the region.

Reply:
👍  Made it configurable (its tied to the location of the db)